### PR TITLE
Background section in PEP 483

### PR DESCRIPTION
**Pull request by ilevkivskyi**
I finished "Background" section, but I think I will make another run to polish the text, when all other sections will be ready. Also I made "Fundamental building blocks" more sparse and fixed several typos.

### CODE REVIEW

File: pep-0483.txt [hunk 1]
```diff
@@ -51,23 +52,127 @@ Notational conventions
...
 
-TODO: Explaint types as sets, subtyping, nominal and structural subtyping.
-Many examples.
+There are many definitions of the concept of type in the literature.
+Here we assume that type is a set of values and a set of functions that
+one can apply to these values. Note that such definition
+could have mathematical problems such as `Russel's paradox
+<https://en.wikipedia.org/wiki/Russell's_paradox>`_
+because the two above sets could be "too infinite".
+Nevertheless it is a good practical mental model for semantics of types.
```
Comment (gvanrossum) [inline pep-0483.txt]: I think this is too academic to bother mentioning and I doubt that any of the readers care about this level of disclaimers.

Comment (ilevkivskyi) [inline pep-0483.txt]: OK, I agree. Maybe later we could add few links for people who want to learn more about types, generics, variance, etc.

File: pep-0483.txt [hunk 2]
```diff
@@ -51,23 +52,127 @@ Notational conventions
...
+
+There are several ways to define a particular type:
```
Comment (gvanrossum) [inline pep-0483.txt]: I think the section below is also a bit too theoretical, but I need to focus on some other things before writing up an alternative or critique. Hopefully in a few days.

**Comment by ilevkivskyi** [comment]
@gvanrossum I pushed a new commit to this PR, I tried to weed out everything unnecessary theoretical, added examples, tried to use more intuitive language. Could you please comment on this new version, when you will have time.

**Comment by gvanrossum** [comment]
Here's what I plan to do. I'll merge this PR locally, then I'll add some small corrections of my own as a separate commit. Then I'll push and close this.

**Comment by ilevkivskyi** [comment]
Could you please wait 5 minutes? I will push one final improvement to make it even less theoretical

**Comment by ilevkivskyi** [comment]
I like this idea. In the meantime I started working on sections about generics and covariance. I will make another PR soon

**Comment by gvanrossum** [comment]
OK, I will use that version. Please don't add more to this PR and please hold off on editing text you added here in the next PR (I am doing an extensive editing pass for style).

**Comment by ilevkivskyi** [comment]
OK, sure.

**Comment by gvanrossum** [comment]
Gotta run, I pushed out what I've done so far. I got to the first paragraph of "Types vs. Classes".

Feel free to add another PR at this point.

**Pull request closed without merging**